### PR TITLE
14930 - check role before remove user from keycloak group

### PR DESCRIPTION
**Pull request by Jxio**
*Issue #:*
https://github.com/bcgov/entity/issues/14930
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/src/auth_api/services/keycloak.py [hunk 1]
```diff
@@ -222,6 +222,9 @@ def remove_from_account_holders_group(keycloak_guid: str = None, **kwargs):
         """Remove user from the group."""
         if not keycloak_guid:
             user_from_context: UserContext = kwargs['user_context']
+            # check if the role 'account_holder' is present.
+            if Role.ACCOUNT_HOLDER.value not in user_from_context.roles:
+                raise BusinessException(Error.MEMBERSHIP_NOT_FOUND, None)
```
Comment (seeker25) [inline auth-api/src/auth_api/services/keycloak.py:227]: This is still going to cause an error when removing staff. They don't have `Role.ACCOUNT_HOLDER`

File: auth-api/src/auth_api/services/keycloak.py [hunk 2]
```diff
@@ -222,6 +222,9 @@ def remove_from_account_holders_group(keycloak_guid: str = None, **kwargs):
...
             keycloak_guid: Dict = user_from_context.sub
 
         KeycloakService._remove_user_from_group(keycloak_guid, GROUP_ACCOUNT_HOLDERS)
```
Comment (seeker25) [inline auth-api/src/auth_api/services/keycloak.py:230]:
```
if Role.ACCOUNT_HOLDER.value in user_from_context.roles:
 KeycloakService._remove_user_from_group(keycloak_guid, GROUP_ACCOUNT_HOLDERS)
```

File: auth-api/src/auth_api/exceptions/errors.py [hunk 3]
```diff
@@ -86,6 +86,8 @@ class Error(Enum):
                                    http_status.HTTP_400_BAD_REQUEST
     PATCH_INVALID_ACTION = 'PATCH_INVALID_ACTION', http_status.HTTP_400_BAD_REQUEST
 
+    MEMBERSHIP_NOT_FOUND = 'The requested membership record could not be found.', http_status.HTTP_404_NOT_FOUND
```
Comment (seeker25) [inline auth-api/src/auth_api/exceptions/errors.py:89]: Remove, see below

File: auth-api/src/auth_api/services/keycloak.py (hunk 1 above)
Comment (seeker25) [reply auth-api/src/auth_api/services/keycloak.py:227]: Remove, see below

File: auth-api/src/auth_api/exceptions/errors.py (hunk 3 above)
Comment (Jxio) [reply auth-api/src/auth_api/exceptions/errors.py]: ok, removed

**Review by seeker25 — APPROVED** [review]

**Comment by seeker25** [comment]
Please fix CI/CD

**Comment by seeker25** [comment]
Please fix CI/CD

**Pull request merged**